contracts/mocks/SimpleServiceManager.sol

```diff
@@ -15,23 +15,32 @@ contract SimpleServiceManager is IWavsServiceManager {
         IWavsServiceHandler.Envelope calldata /* envelope */,
         IWavsServiceHandler.SignatureData calldata signatureData
     ) external view override {
-        // Validate that operators are sorted in ascending byte order
-        require(
-            _validateOperatorSorting(signatureData.signers),
-            "Operators are not properly sorted"
-        );
+        // Input validation
+        if (signatureData.signers.length == 0 || signatureData.signers.length != signatureData.signatures.length) {
+            revert IWavsServiceManager.InvalidSignatureLength();
+        }
+        if (signatureData.referenceBlock >= block.number) {
+            revert IWavsServiceManager.InvalidSignatureBlock();
+        }
+        if (!_validateOperatorSorting(signatureData.signers)) {
+            revert IWavsServiceManager.InvalidSignatureOrder();
+        }
 
         // Get the total operator weight of these signatures
         uint256 totalWeight = 0;
         for (uint256 i = 0; i < signatureData.signers.length; i++) {
             totalWeight += operatorWeights[signatureData.signers[i]];
         }
 
-        // Check if total weight is above threshold
-        require(
-            totalWeight >= lastCheckpointThresholdWeight,
-            "Not enough operator weight"
-        );
+        // Avoid 0 weight ever passing this check
+        if (totalWeight == 0) {
+            revert IWavsServiceManager.InsufficientQuorum();
+        }
```

Comment:
Should the error add extra info to say that totalWeight is 0? -- or maybe a new error type IWavsServiceManager.InsufficientQuorumIsZero();

Reply:
done in https://github.com/Lay3rLabs/wavs-middleware/pull/132/commits/c71c9707fc298490b0b5cdee9abd022fbcaa00c7